src/components/BaseKeymap.vue

```diff
@@ -4,12 +4,8 @@ export default {
   methods: {
     calcKeyKeymapDims(w, h) {
       return {
-        w:
-          w * this.config.KEY_X_SPACING -
-          (this.config.KEY_X_SPACING - this.config.KEY_WIDTH),
-        h:
-          h * this.config.KEY_Y_SPACING -
-          (this.config.KEY_Y_SPACING - this.config.KEY_HEIGHT),
+        w: (w-1) * this.config.KEY_X_SPACING + this.config.KEY_WIDTH,
```

Comment:
try using a really big keyboard layout. this code doesn't do what you think it does. by using a constant instead of the configuration block you have probably introduced a bug. It's a scaler.

Reply:
So, I'm trying to understand what the advantage of this is? Maybe I'm too dense, but when I punch numbers into a spread sheet they no longer seem equivalent.

w = 3, KEY_X_SPACING = 40, KEY_WIDTH = 45.
Old formula = (3 * 40) - (40-45) = 157
New formula = 2 * 40 + 45 = 125

Unless I'm missing something, this holds true for all values of w.

Reply:
I read:

```python
rslt = w * this.config.KEY_X_SPACING - (this.config.KEY_X_SPACING - this.config.KEY_WIDTH)
rslt = w * this.config.KEY_X_SPACING - this.config.KEY_X_SPACING + this.config.KEY_WIDTH
rslt = (w - 1) * this.config.KEY_X_SPACING + this.config.KEY_WIDTH
```
That seems the same to me

```python
w = 3, KEY_X_SPACING = 45, KEY_WIDTH = 40.
# before
rslt = w * this.config.KEY_X_SPACING - (this.config.KEY_X_SPACING - this.config.KEY_WIDTH)
Old formula = (3 * 45) - (45-40) = 130

# after
rslt = (w - 1) * this.config.KEY_X_SPACING + this.config.KEY_WIDTH
New formula = 2 * 45 + 40 = 130
```

Reply:
I see. I punched in the wrong value. My bad.